woocommerce/woocommerce
# [e2e tests] Fix more e2e tests with Gutenberg active

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

More fixes for tests that are failing with Gutenberg installed.
Closes #46868 
Follow up of #46842 which missed some of the issues.

- updated multiple tests to use the getCanvas util instead of page, to ensure the editor canvas iframe introduced by Gutenberg is detected.
- created more utilities to help reducing duplication and ease the maintenance burden.
- fixed some eslint errors, most importantly removed the wait for `networkidle` when adding a product to basket and replaced it with waiting for the relevant request instead.
- made some of the test page names to be unique. this ensures that if the first test (which is a dependency to the next ones) fails the other tests are also failing, instead of using another possibly existing page, which is not intended. Used `uuid` instead to `Date.now()`, because the timestamp can contain the same numbers sequence as prices and make some tests fail. As a side note, the tests dependencies should be addressed with future PRs, this is not an ideal test design.





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check all affected tests with and without Gutenberg installed.
2. CI should be green. See runs: [with Gutenberg](https://github.com/woocommerce/woocommerce/actions/runs/8830456847/job/24244662867?pr=46861), without Gutenberg.
